Initial commit

beechtom opened the pull request.

beechtom committed:
Initial commit

lucywyman commented on metadata.json:
```diff
@@ -0,0 +1,88 @@
+  "summary": "Run a command or script with sensitive environment variables.",
+  "license": "Apache-2.0",
+  "source": "https://github.com/puppetlabs/puppetlabs-secure_env_vars",
+  "dependencies": [
```
We should list puppetlabs-stdlib here if we're using parsejson

beechtom replied:
Added as a dependency. Not really sure how open to be on the version requirement, so I just picked `>= 6.0.0`.

lucywyman commented on Gemfile:
```diff
@@ -0,0 +1,72 @@
+minor_version = ruby_version_segments[0..1].join('.')
+
+group :development do
+  gem "fast_gettext", '1.1.0',                                   require: false if Gem::Version.new(RUBY_VERSION.dup) < Gem::Version.new('2.1.0')
```
What do we need all these gems for? We should only need Puppet, and maybe hiera and facter

beechtom replied:
It's just the default for `pdk module new`.

lucywyman commented on README.md:
````diff
@@ -0,0 +1,95 @@
+_\*nix shell command_
+
+```shell
+bolt plan run secure_env_vars targets=servers command=somecommand
````
```suggestion
bolt plan run secure_env_vars targets=servers command="git pull https://$OAUTH_TOKEN:x-oauth-basic@github.com/name/repo.git master"
```
Or some other more real example

lucywyman commented on README.md:
```diff
@@ -0,0 +1,95 @@
+
+### Running a script
+
+To run a script, set the `script` parameter:
```
It'd be useful to point out this can either be a relative path, absolute path, or file from a module on the modulepath

lucywyman commented on Rakefile:
```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
```
Can we just manually manage the changelog and not include any of this? It should just be:
```
require 'puppetlabs_spec_helper/rake_tasks'
require 'puppet_blacksmith/rake_tasks' if Bundler.rubygems.find_name('puppet-blacksmith').any?
```

lucywyman commented on metadata.json:
```diff
@@ -0,0 +1,92 @@
+  "dependencies": [
+    {
+      "name": "puppetlabs-stdlib",
+      "version_requirement": ">= 6.0.0"
```
This seems high, particularly since many modules have a maximum version of 5.x - I think we can go back to maybe 4.24, which is https://github.com/puppetlabs/puppetlabs-stdlib/commit/21e7ce8fc3cdacab50b71d1973d7c80ed0543fd3#diff-a03c09e759361331222b5070179514a282f9b3a949b91f133f364b309d20a189

lucywyman reviewed (commented):
In looking at the other modules we ship, most have some spec testing. I think it'd be good to add tests to verify that passing script and command fails, passing neither fails, and that passing one of each succeeds.

beechtom commented:
Added spec tests and GitHub workflow. Also updated the plan to handle `BOLT_ENV_VARS` being unset, as passing an `Undef` value to `parsejson` results in an error.
